### DISCUSSION TITLE
Windows compilation error

### DISCUSSION THREAD

**Discussion by zanxueyan** [cyberbotics/webots#4912] [Q&A]
The following error occurs when the last step of make-j12 is executed:
`MINGW64 /d/webots
# make -j12
Makefile:58: `pwd -W | tr -s / '//'`/resources/Makefile.os.include: No such file or directory
make: *** No rule to make target '`pwd -W | tr -s / '//'`/resources/Makefile.os.include'.  Stop.
`
What should I do? Please give guidance, thank you!

**Comment by omichel** [comment]
Did you follow precisely the instructions from https://github.com/cyberbotics/webots/wiki/Windows-installation/?

**Comment by zanxueyan** [comment]
![1658228260(1)](https://user-images.githubusercontent.com/33081904/179734581-18eed754-d0a0-4946-972a-df90d3b81a02.png)
Yes,I follow the documentation steps exactly until I get an error as shown in the figure

**Comment by zanxueyan** [comment]
Since I am not familiar with Linux, I would like to use Visual Studio for debugging and compilation under Windows.Can I achieve this result?

**Reply by omichel** [reply]
Unfortunately, we do not use Visual Studio to compile Webots. Instead, we use gcc (which comes with MSYS2 on Windows). But it is possible to compile robot controllers using Visual Studio.

**Reply by zanxueyan** [reply]
Thanks

**Comment by zanxueyan** [comment]
![86279f5594b11482cc5dbc0bc7d495d](https://user-images.githubusercontent.com/33081904/179750882-8bbe3ba5-592d-45db-b8e5-81e1e8315713.png)
I operated according to the steps again, and there was no mistake last time, but there was a new mistake

**Comment by zanxueyan** [comment]
Modified the code in generate Proto list.py as shown below，OK
![1658235007(1)](https://user-images.githubusercontent.com/33081904/179754321-87cee010-fc77-42c1-8719-c5430e7e1260.png)

**Reply by omichel** [reply]
Strange, this not needed on my computer. May I ask you to type "which python" and "python -VV" to determine which version of python your are using?

**Reply by zanxueyan** [reply]
# python -VV
Python 3.10.5 (main, Jun 18 2022, 01:16:30)  [GCC 12.1.0 64 bit (AMD64)]

**Comment by zanxueyan** [comment]
# python -VV
Python 3.10.5 (main, Jun 18 2022, 01:16:30)  [GCC 12.1.0 64 bit (AMD64)]

**Reply by omichel** [reply]
May I ask you also to give me the output of the following:
```bash
printenv LANG
printenv LC_CTYPE
```
I have:
```
LANG = en_US.UTF-8
LC_CTYPE = en_GB.UTF-8
```

Also, can you try to set:
```
export PYTHONIOENCODING=UTF-8
```
before you start the script?

**Comment by Kreijstal** [comment]
> The following error occurs when the last step of make-j12 is executed: `MINGW64 /d/webots
> # make -j12
> 
> Makefile:58: `pwd -W | tr -s / '//'`/resources/Makefile.os.include: No such file or directory make: *** No rule to make target '`pwd -W | tr -s / '//'`/resources/Makefile.os.include'. Stop. ` What should I do? Please give guidance, thank you!

I had this error before, it seems to happen when you dont source https://github.com/cyberbotics/webots/blob/master/scripts/install/bash_profile.windows within mingw.. I still dont understand why

**Reply by omichel** [reply]
That is likely because the `bash_profile.windows` is calling in turn https://github.com/cyberbotics/webots/blob/master/scripts/install/msys.sh which changes the `PATH` environment variable to include the msys2 binaries ahead of the default windows binaries.

**Comment by omichel** [comment]
The problem should be fixed in #4982.
